huangjoyce3/MealMatchup
# Emergency Pickup Schema

## Emergency Pickup Schema
Schema change to prepare for the new Emergency Pickup feature.

### Changes
- Updated `schema.txt` to reflect the new schema. Overview:
  - added `type` field to both `delivery_request` and `delivery`, which is either `DeliveryType.RECURRING` or `DeliveryType.EMERGENCY`
  - annotated what fields could be null depending on what pickup type it is
- Update current `delivery_request` and `delivery` creation for recurring pickups to include the new field
- Added `NotificationType`s for emergency pickups
- Manually added `type` field to all `delivery_request`s and `delivery`s in the PROD DB
- Misc:
  - synced `functions/Enums.js` with `src/Enums.js`
  - renamed `NotificationCategory` to `NotificationContentType`
  - make sure usage of `delivery.deliverers` don't depend on `length=2` since now we could have `length=1` for emergency pickups (RA driver)
  - remove `isEmergency` from `MobileDelivery` (the only place that references that field)

### Testing
- created delivery requests, inspect db
- deploy functions and created deliveries, inspect db
- make sure notification still works
- make sure calendar DG Info still works
- make sure Mobile Delivery still works
